sql/src/SqlTables.sk

```diff
@@ -357,6 +357,176 @@ fun getLatestIndexDescr(
   }
 }
 
+// Recompile and reevaluate any views that depend upon any of the altered directories
+private fun resetReactiveViews(
+  context: mutable SKStore.Context,
+  alteredDirs: SortedSet<P.Name>,
+  options: Options,
+): void {
+  viewsDir = context.unsafeGetEagerDir(getViewsDir(context).dirName);
+  viewsToReset = viewsDir
+    .unsafeGetFileIter()
+    .flatMap((x) ->
+      x.i1.filter((file) ->
+        SelectFile::type(file).cselect.from.any((from) ->
+          alteredDirs.contains(from.i0.name)
+        )
+      )
+    )
+    .collect(Array);
+
+  for (childView in viewsToReset) {
+    selectFile = SelectFile::type(childView);
+    newCSelect = SKDB.Compiler::create(
+      false,
+      options with {reactive => true},
+      selectFile.select.pos,
+      selectFile.queryParams,
+    ).compileSelect(context, selectFile.select, selectFile.name, true);
+    dirName = SKStore.DirName::create("/" + selectFile.name + "/");
+
+    tableDir = context.unsafeGetEagerDir(getTableDir(context).dirName);
+    fileName = SKStore.SID::create(selectFile.name.lower);
+    tableDir.getArrayRaw(fileName).each((x) -> {
+      table = context.unsafeGetEagerDir(DirDescr::type(x).dirName);
+      clearData: (
+        (SKStore.Key, mutable Iterator<SKStore.File>),
+      ) -> mutable Iterator<
+        (SKStore.Key, (SKStore.Path, SKStore.Path, Array<SKStore.File>)),
+      > = (row) -> {
+        key = row.i0;
+        for (datum in table.unsafeGetAllDataIter(key)) {
+          (_tick, source, writer, _value) = datum;
+          yield (key, (source, writer, Array[]))
+        }
+      };
+      table.writeArraySourceMany(
+        context,
+        table.unsafeGetFileIter().flatMap(clearData),
```

Comment:
Why do we need to clear the data first? Can we not just build a new dir from the evalSelect and replace the current? I guess there's some hassle with setting up the downstream views?

Reply:
Yes, it's to do with the downstream dependencies.  I originally built a new dir and tried to rewire things, but it was pretty hairy and Julien suggested doing it this way instead.